lib/syntax.ml

```diff
@@ -108,9 +108,15 @@ let exists_grouping columns =
 
 (* all columns from tables, without duplicates *)
 (* FIXME check type of duplicates *)
-let make_unique = List.unique ~cmp:Schema.Source.Attr.(fun a1 a2 -> a1.attr.name = a2.attr.name && a1.attr.name <> "")
+let make_unique =
+  List.unique ~cmp:(fun a1 a2 ->
+      (* Check if alias has same name as column OR if columns are from the same table (source)  *)
+      (a2.Schema.Source.Attr.sources = [] || a1.Schema.Source.Attr.sources = a2.sources)
+      && a1.attr.name = a2.attr.name
+      (* Check if the tested column is unnamed (not a column) and unaliased *)
```

Comment:
btw this comment is not correct (and related to my previous comment about this function becoming non-symmetrical)
what it is testing is that both columns are not unnamed (because of comparison above), there is no such thing as "tested column"

Reply:
Fixed, now is symmetrical. 

```sql
  tt "SELECT COUNT(column_a) as column_a FROM test23 WHERE column_a = @column_a" [
    (* COUNT(column_a :: Text) :: Int *)
    attr' "column_a" Int;
  ] [
    named "column_a" Text;
  ];
```
I was keeping it: `a2.Schema.Source.Attr.sources = []` , but realized that WHERE doesn't support aliases. Removed now

Reply:
Updated table with example

Reply:
```sql
SELECT MAX(id) as id2
FROM test21
where id2 > 0;
```
This isn't valid